### PR TITLE
Add scripts for "factory containers"

### PR DESCRIPTION
**Pull request by doanac**
These are the scripts I've been using for building multi-arch containers in CI and then having their corresponding docker-app(s) published into an OTA targets.json.

### CODE REVIEW

**Comment by doanac** [comment]
Here's an example of it in action: https://ci.foundries.io/projects/andy-corp/lmp/builds/86/

File: factory-containers/build.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#!/bin/sh -e
```
Comment (ricardosalveti) [inline factory-containers/build.sh]: $ checkbashisms factory-containers/build.sh
possible bashism in factory-containers/build.sh line 5 (should be '.', not 'source'):
source $HERE/../helpers.sh
possible bashism in factory-containers/build.sh line 40 (${foo:3[:1]}):
TAG=${GIT_SHA:0:6}

If not changed to be generic, maybe force bash instead?

File: factory-containers/docker-app-publish.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/sh -e
```
Comment (ricardosalveti) [inline factory-containers/docker-app-publish.sh]: $ checkbashisms factory-containers/docker-app-publish.sh
possible bashism in factory-containers/docker-app-publish.sh line 5 (should be '.', not 'source'):
source $HERE/../helpers.sh
possible bashism in factory-containers/docker-app-publish.sh line 10 (${foo:3[:1]}):
TAG=${GIT_SHA:0:6}
possible bashism in factory-containers/docker-app-publish.sh line 25 (alternative test command ([[ foo ]] should be [ foo ])):
if [[ "$sha" = "$newsha" ]] && [[ -n "$apps" ]] ; then

Same here.

File: factory-containers/test_build.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -eu
```
Comment (ricardosalveti) [inline factory-containers/test_build.sh]: This one doesn't need to use bash directly :-)

**Comment by ricardosalveti** [comment]
Mind also adding: Copyright (c) 2019 Foundries.io, SPDX-License-Identifier: Apache-2.0 to the .sh files?

File: factory-containers/build.sh (hunk 1 above)
Comment (doanac) [reply factory-containers/build.sh]: its got to bin /bin/sh - docker:dind doesn't have bash.

**Comment by doanac** [comment]
I think the comments are resolved. I'm testing the actual code here: 
  https://ci.foundries.io/projects/andy-corp/lmp/builds/88/

**Comment by doanac** [comment]
i decided to keep with manifest-tool for now. Its a bit more invasive to get into our code base at the moment. We should look at just moving the whole repo over to it in the future.

**Comment by ricardosalveti** [comment]
> i decided to keep with manifest-tool for now. Its a bit more invasive to get into our code base at the moment. We should look at just moving the whole repo over to it in the future.

Sounds good.

LGTM, just not so familiar with the codebase for more in depth review.

**Pull request merged**